leshan-core/src/main/java/org/eclipse/leshan/core/node/codec/senml/LwM2mNodeSenMLJsonDecoder.java

```diff
@@ -0,0 +1,327 @@
+                    recordsByInstanceId.values().iterator().next(), baseName, model);
+
+            // validate there is only 1 resource
+            if (resourcesMap.size() != 1)
+                throw new CodecException("One resource should be present in the payload [path:%s]", path);
+
+            node = resourcesMap.values().iterator().next();
+        } else {
+            throw new IllegalArgumentException("invalid node class: " + nodeClass);
+        }
+
+        return (T) node;
+    }
+
+    /**
+     * Extract and validate base name from SenML pack
+     * 
+     * @param pack
+     * @param requestPath
+     * @return
+     * @throws CodecException
+     */
+    private static LwM2mPath extractAndValidateBaseName(SenMLPack pack, LwM2mPath requestPath) throws CodecException {
+        String baseName = null;
+        for (SenMLRecord record : pack.getRecords()) {
+            if (record.getBaseName() != null && !record.getBaseName().isEmpty()) {
+                baseName = record.getBaseName();
+                break;
+            }
+        }
+
+        // Check baseName is valid
+        if (baseName != null && !baseName.isEmpty()) {
+            LwM2mPath bnPath = new LwM2mPath(baseName);
+
+            // check returned base name path is under requested path
+            if (requestPath.getObjectId() != null && bnPath.getObjectId() != null) {
+                if (!bnPath.getObjectId().equals(requestPath.getObjectId())) {
+                    throw new CodecException("Basename path [%s] does not match requested path [%s].", bnPath,
+                            requestPath);
+                }
+                if (requestPath.getObjectInstanceId() != null && bnPath.getObjectInstanceId() != null) {
+                    if (!bnPath.getObjectInstanceId().equals(requestPath.getObjectInstanceId())) {
+                        throw new CodecException("Basename path [%s] does not match requested path [%s].", bnPath,
+                                requestPath);
+                    }
+                    if (requestPath.getResourceId() != null && bnPath.getResourceId() != null) {
+                        if (!bnPath.getResourceId().equals(requestPath.getResourceId())) {
+                            throw new CodecException("Basename path [%s] does not match requested path [%s].", bnPath,
+                                    requestPath);
+                        }
+                    }
+                }
+            }
+            return bnPath;
+        }
+        return null;
+    }
+
+    /**
+     * Group all SenML record by instanceId
+     * 
+     * @param records
+     * @param baseName
+     *
+     * @return a map (instanceId => collection of SenML Record)
+     */
+    private static Map<Integer, Collection<SenMLRecord>> groupRecordsByInstanceId(Collection<SenMLRecord> records,
+            LwM2mPath baseName) throws CodecException {
+        Map<Integer, Collection<SenMLRecord>> result = new HashMap<>();
+
```

Comment:
Following comment above, I feel that we don't need baseName parameter anymore.
And we should just use `baseName` from `SenMLRecord`.
Look at this part of the rfc to implement it correctly : https://tools.ietf.org/html/rfc8428#section-5.1.6

Reply:
Same as above comments, I am trying to update value of name field of each of record to full path, then base name can be ignore, any comments for this change?